From d5ea2625beb6c1d21ed289090e865234bab3e0bf Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Tue, 9 Jun 2020 10:53:17 +0800
Subject: [PATCH 28/60] waylandsink: add wl_registry.global_remove listener

when hotplug display, wayland client will call this listener
to notify client do clean up. Temporarily set a dummy function
here to avoid app abort

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/1327>
(cherry picked from commit d77de13ca6028d0011f5a2a285a2a6f751b5d8da)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/wldisplay.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/ext/wayland/wldisplay.c b/ext/wayland/wldisplay.c
index 940009556..f7115d59f 100644
--- a/ext/wayland/wldisplay.c
+++ b/ext/wayland/wldisplay.c
@@ -234,8 +234,16 @@ registry_handle_global (void *data, struct wl_registry *registry,
   }
 }
 
+static void
+registry_handle_global_remove (void *data, struct wl_registry *registry,
+    uint32_t name)
+{
+  /* temporarily do nothing */
+}
+
 static const struct wl_registry_listener registry_listener = {
-  registry_handle_global
+  registry_handle_global,
+  registry_handle_global_remove
 };
 
 static gpointer
-- 
2.20.1

